shell/platform/android/io/flutter/view/FlutterMain.java

```diff
@@ -265,40 +272,48 @@ public void run() {
     private static native void nativeInit(Context context, String[] args, String bundlePath, String appStoragePath, String engineCachesPath);
     private static native void nativeRecordStartTimestamp(long initTimeMillis);
 
-    /**
-     * Initialize our Flutter config values by obtaining them from the
-     * manifest XML file, falling back to default values.
-     */
-    private static void initConfig(Context applicationContext) {
+    @NonNull
+    private static ApplicationInfo getApplicationInfo(@NonNull Context applicationContext) {
         try {
-            Bundle metadata = applicationContext.getPackageManager().getApplicationInfo(
-                applicationContext.getPackageName(), PackageManager.GET_META_DATA).metaData;
-            if (metadata != null) {
-                sAotSharedLibraryPath = metadata.getString(PUBLIC_AOT_AOT_SHARED_LIBRARY_PATH, DEFAULT_AOT_SHARED_LIBRARY_PATH);
-                sAotVmSnapshotData = metadata.getString(PUBLIC_AOT_VM_SNAPSHOT_DATA_KEY, DEFAULT_AOT_VM_SNAPSHOT_DATA);
-                sAotVmSnapshotInstr = metadata.getString(PUBLIC_AOT_VM_SNAPSHOT_INSTR_KEY, DEFAULT_AOT_VM_SNAPSHOT_INSTR);
-                sAotIsolateSnapshotData = metadata.getString(PUBLIC_AOT_ISOLATE_SNAPSHOT_DATA_KEY, DEFAULT_AOT_ISOLATE_SNAPSHOT_DATA);
-                sAotIsolateSnapshotInstr = metadata.getString(PUBLIC_AOT_ISOLATE_SNAPSHOT_INSTR_KEY, DEFAULT_AOT_ISOLATE_SNAPSHOT_INSTR);
-                sFlutterAssetsDir = metadata.getString(PUBLIC_FLUTTER_ASSETS_DIR_KEY, DEFAULT_FLUTTER_ASSETS_DIR);
-            }
+            return applicationContext
+                .getPackageManager()
+                .getApplicationInfo(applicationContext.getPackageName(), PackageManager.GET_META_DATA);
         } catch (PackageManager.NameNotFoundException e) {
             throw new RuntimeException(e);
         }
     }
 
-    private static void initResources(Context applicationContext) {
-        Context context = applicationContext;
-        new ResourceCleaner(context).start();
+    /**
+     * Initialize our Flutter config values by obtaining them from the
+     * manifest XML file, falling back to default values.
+     */
+    private static void initConfig(@NonNull Context applicationContext) {
+        Bundle metadata = getApplicationInfo(applicationContext).metaData;
 
-        Bundle metaData = null;
-        try {
-            metaData = context.getPackageManager().getApplicationInfo(
-                    context.getPackageName(), PackageManager.GET_META_DATA).metaData;
+        if (metadata == null) {
```

Comment:
Can you add a comment as to why metadata would be null? Does that signify empty metadata? or a failure? or something else?

Reply:
Done.